sirepo/template/cloudmc.py

```diff
@@ -197,14 +197,117 @@ def stateless_compute_validate_material_name(data, **kwargs):
 
 
 def write_parameters(data, run_dir, is_parallel):
-    pkio.write_text(
-        run_dir.join(template_common.PARAMETERS_PYTHON_FILE),
-        _generate_parameters_file(data, run_dir=run_dir),
-    )
+    if _is_sbatch_run_mode(data):
+        pkio.write_text(
+            run_dir.join(f"{_PREP_SBATCH_PREFIX}.py"),
+            _generate_parameters_file(data, run_dir=run_dir),
+        )
+        with pkio.open_text(run_dir.join(f"{_PREP_SBATCH_PREFIX}.log"), mode="w") as l:
+            subprocess.run(
+                ["python", f"{_PREP_SBATCH_PREFIX}.py"],
+                check=True,
+                stderr=l,
+                stdout=l,
+            )
+        pkio.write_text(
+            run_dir.join(template_common.PARAMETERS_PYTHON_FILE),
+            template_common.render_jinja(
```

Comment:
I don't know whether this gets executed right away or waits for the subprocess. Does it depend on the results?

Reply:
Yes, subprocess.run() waits for the command to complete before continuing.